mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -2195,6 +2214,128 @@ impl MmCoin for TendermintCoin {
     fn on_disabled(&self) -> Result<(), AbortedError> { AbortableSystem::abort_all(&self.abortable_system) }
 
     fn on_token_deactivated(&self, _ticker: &str) {}
+
+    async fn handle_balance_stream(self, ctx: MmArc) {
+        fn generate_subscription_query(query_filter: String) -> String {
+            let q = json!({
+                "jsonrpc": "2.0",
+                "method": "subscribe",
+                "id": 0,
+                "params": {
+                    "query": query_filter
+                }
+            });
+
+            q.to_string()
+        }
+
+        let account_id = self.account_id.to_string();
+        let mut current_balances: HashMap<String, BigDecimal> = HashMap::new();
+
+        let receiver_q = generate_subscription_query(format!("coin_received.receiver = '{}'", account_id));
+        let receiver_q = tokio_tungstenite_wasm::Message::Text(receiver_q);
+
+        let spender_q = generate_subscription_query(format!("coin_spent.spender = '{}'", account_id));
+        let spender_q = tokio_tungstenite_wasm::Message::Text(spender_q);
+
+        loop {
+            let node_uri = match self.rpc_client().await {
+                Ok(client) => client.uri(),
+                Err(e) => {
+                    error!("{e}");
+                    continue;
+                },
+            };
+
+            let socket_address = format!("{}/{}", http_uri_to_ws_address(node_uri), "websocket");
+
+            let mut wsocket = match tokio_tungstenite_wasm::connect(socket_address).await {
+                Ok(ws) => ws,
+                Err(e) => {
+                    error!("{e}");
+                    continue;
+                },
+            };
+
+            // Filter received TX events
+            if let Err(e) = wsocket.send(receiver_q.clone()).await {
+                error!("{e}");
+                continue;
+            }
+
+            // Filter spent TX events
+            if let Err(e) = wsocket.send(spender_q.clone()).await {
+                error!("{e}");
+                continue;
+            }
+
+            while let Some(message) = wsocket.next().await {
+                let msg = match message {
+                    Ok(tokio_tungstenite_wasm::Message::Text(data)) => data.clone(),
+                    Ok(tokio_tungstenite_wasm::Message::Close(_)) => break,
+                    Err(err) => {
+                        error!("{err}");
```

Comment:
this error might not be informative enough, so I think it's better we add additional information in this log e.g
```log
error!("Server returned an unknown message type - {err}")
```